spec/unit/plugins/windows/uptime_spec.rb

```diff
@@ -0,0 +1,88 @@
+#
+# Author:: Aliasgar Batterywala (<aliasgar.batterywala@msystechnologies.com>)
+# Copyright:: Copyright (c) 2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require File.expand_path(File.dirname(__FILE__) + "/../../../spec_helper.rb")
+
+describe Ohai::System, "Windows plugin uptime" do
+
+  let(:wmi) { double("wmi", { :first_of => "" }) }
+
+  before(:each) do
+    allow(WmiLite::Wmi).to receive(:new).and_return(wmi)
+  end
+
+  ## Windows newer versions category here includes server OS starting from Windows Server 2008 ##
+  shared_context "WMI class to use to fetch system uptime for newer versions of Windows platform" do
```

Comment:
can you make the names of the shared context's more brief? that's a lot of words with only one in the middle that is different. It should be easier to tell the difference just looking at them.